be/src/vec/exec/format/table/parquet_metadata_reader.cpp

```diff
@@ -0,0 +1,908 @@
+    META_DISTINCT_COUNT,
+    META_ENCODINGS,
+    META_COMPRESSION,
+    META_DATA_PAGE_OFFSET,
+    META_INDEX_PAGE_OFFSET,
+    META_DICTIONARY_PAGE_OFFSET,
+    META_TOTAL_COMPRESSED_SIZE,
+    META_TOTAL_UNCOMPRESSED_SIZE,
+    META_STATISTICS_MIN,
+    META_STATISTICS_MAX,
+    META_COLUMN_COUNT
+};
+
+constexpr std::array<const char*, SCHEMA_COLUMN_COUNT> kSchemaColumnNames = {
+        "file_name",        "column_name",       "column_path",       "physical_type",
+        "logical_type",     "repetition_level",  "definition_level",  "type_length",
+        "precision",        "scale",             "is_nullable"};
+
+static_assert(kSchemaColumnNames.size() == SCHEMA_COLUMN_COUNT);
+
+constexpr std::array<const char*, META_COLUMN_COUNT> kMetadataColumnNames = {
+        "file_name", "row_group_id", "column_id", "column_name", "column_path", "physical_type",
+        "logical_type", "type_length", "converted_type", "num_values", "null_count",
+        "distinct_count", "encodings", "compression", "data_page_offset", "index_page_offset",
+        "dictionary_page_offset", "total_compressed_size", "total_uncompressed_size",
+        "statistics_min", "statistics_max"};
+
+static_assert(kMetadataColumnNames.size() == META_COLUMN_COUNT);
+
+std::string join_path(const std::vector<std::string>& items) {
+    return join(items, ".");
+}
+
+template <typename ColumnType, typename T>
+void insert_numeric_impl(MutableColumnPtr& column, T value) {
+    if (auto* nullable = check_and_get_column<ColumnNullable>(*column)) {
+        nullable->get_null_map_data().push_back(0);
+        auto& nested = nullable->get_nested_column();
+        assert_cast<ColumnType&>(nested).insert_value(value);
+    } else {
+        assert_cast<ColumnType&>(*column).insert_value(value);
+    }
+}
+
+void insert_int32(MutableColumnPtr& column, Int32 value) {
+    insert_numeric_impl<ColumnInt32>(column, value);
+}
+
+void insert_int64(MutableColumnPtr& column, Int64 value) {
+    insert_numeric_impl<ColumnInt64>(column, value);
+}
+
+void insert_bool(MutableColumnPtr& column, bool value) {
+    insert_numeric_impl<ColumnUInt8>(column, static_cast<UInt8>(value));
+}
+
+void insert_string(MutableColumnPtr& column, const std::string& value) {
+    if (auto* nullable = check_and_get_column<ColumnNullable>(*column)) {
+        nullable->get_null_map_data().push_back(0);
+        auto& nested = nullable->get_nested_column();
+        assert_cast<ColumnString&>(nested).insert_data(value.c_str(), value.size());
+    } else {
+        assert_cast<ColumnString&>(*column).insert_data(value.c_str(), value.size());
+    }
+}
+
+void insert_null(MutableColumnPtr& column) {
+    if (auto* nullable = check_and_get_column<ColumnNullable>(*column)) {
+        nullable->get_null_map_data().push_back(1);
+        nullable->get_nested_column().insert_default();
+    } else {
+        column->insert_default();
+    }
+}
+
+std::string physical_type_to_string(tparquet::Type::type type) {
```

Comment:
extract these `xx_to_xx` methods to a new file `parquet_utils.h`. So that they can be used in other places, maybe.